tink/client_test/hardware_test.go

```diff
@@ -23,13 +23,14 @@ import (
 	. "github.com/onsi/gomega"
 
 	"github.com/tinkerbell/cluster-api-provider-tinkerbell/tink/client"
+	"github.com/tinkerbell/cluster-api-provider-tinkerbell/tink/client/fake"
 	testutils "github.com/tinkerbell/cluster-api-provider-tinkerbell/tink/test/utils"
 )
 
 func TestHardwareLifecycle(t *testing.T) { //nolint:funlen,paralleltest
 	g := NewWithT(t)
 	ctx := context.Background()
-	hardwareClient := client.NewHardwareClient(realHardwareClient(t))
+	hardwareClient := fake.NewFakeHardwareClient()
```

Comment:
I'm not sure we should use the fake client here for testing, the idea was to provide the fake client for other components to use for testing, but we should likely still be testing the real client here for the purpose of this package.

I'm not sure if there is a way to record/replay traffic with grpc in a way that is similar to what can be done with http, but that could be another alternative for testing this package that wouldn't require a live tinkerbell server be up and running.